core/bot_framework_adapter_test.go

```diff
@@ -20,18 +20,59 @@
 package core_test
 
 import (
+	"bytes"
 	"context"
+	"encoding/json"
 	"fmt"
 	"log"
+	"net/http"
+	"net/http/httptest"
 	"os"
+	"testing"
 
 	"github.com/infracloudio/msbotbuilder-go/core"
 	"github.com/infracloudio/msbotbuilder-go/core/activity"
 	"github.com/infracloudio/msbotbuilder-go/schema"
+	"github.com/stretchr/testify/assert"
 )
 
-func Example() {
+func serverMock() *httptest.Server {
+	handler := http.NewServeMux()
+	handler.HandleFunc("/v3/conversations/abcd1234/activities", msTeamsMockMock)
 
+	srv := httptest.NewServer(handler)
+
+	return srv
+}
+
+func msTeamsMockMock(w http.ResponseWriter, r *http.Request) {
+	_, _ = w.Write([]byte("{\"id\":\"1\"}"))
+}
+
+// Create a handler that defines operations to be performed on respective events.
+// Following defines the operation to be performed on the 'message' event.
+var customHandler = activity.HandlerFuncs{
+	OnMessageFunc: func(turn *activity.TurnContext) (schema.Activity, error) {
+		return turn.SendActivity(activity.MsgOptionText("Echo: " + turn.Activity.Text))
+	},
+}
+
+func processMessage(w http.ResponseWriter, req *http.Request) {
+	ctx := context.Background()
+	setting := core.AdapterSetting{
+		AppID:       "asdasd",
+		AppPassword: "cfg.MicrosoftTeams.AppPassword",
+	}
+	adapter, err := core.NewBotAdapter(setting)
+	act, err := adapter.ParseRequest(ctx, req)
+	err = adapter.ProcessActivity(ctx, act, customHandler)
```

Comment:
Check and assert returned errors here

Reply:
done

Reply:
Logging is not enough. We should fail the test if the error is returned by any method